subprojects/language-native/src/main/java/org/gradle/language/swift/tasks/SwiftCompile.java

```diff
@@ -73,4 +75,18 @@ public void setModuleName(String moduleName) {
     public void setModuleName(Provider<String> moduleName) {
         this.moduleName.set(moduleName);
     }
+
+    @Override
+    public void compile(IncrementalTaskInputs inputs) {
+        SimpleStaleClassCleaner cleaner = new SimpleStaleClassCleaner(getOutputs());
```

Comment:
Would you mind renaming `SimpleStaleClassCleaner` to something which covers this usage, too? I think `SimpleStaleOutputCleaner` could be a good idea.
Note that the cleaner does not clean up any directories. If you are sure that you have no overlapping outputs, then you could just delete the output directory and you won't need the complicated logic as in the cleaner.
We use a similar logic in [SkipEmptySourceFilesTaskExecuter](https://github.com/gradle/gradle/blob/f088cebac9b062b67842f521d274dfdcf6b72392/subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/SkipEmptySourceFilesTaskExecuter.java#L72-L83). It would be much better that a task could declare that stale files should be cleaned up instead of adding this cleaner on a case by case basis.